File: week07/wordcount_scala/src/main/scala/wc/WordCountMapper.scala
package wc

import java.util.StringTokenizer

import org.apache.hadoop.io.{IntWritable, Text}
import org.apache.hadoop.mapreduce.Mapper


class WordCountMapper extends Mapper[Object, Text, Text, IntWritable] {
  val word = new Text()
  val one = new IntWritable(1)

  override def map(key: Object, value: Text, context: Mapper[Object, Text, Text, IntWritable]#Context): Unit = {
    val st = new StringTokenizer(value.toString)
    


  }
}

File: week07/wordcount_java/src/main/java/wc/WordCountMapper.java
package wc;

import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Mapper;

import java.io.IOException;
import java.util.StringTokenizer;

public class WordCountMapper extends Mapper<Object, Text, Text, IntWritable> {
  Text word = new Text();
  IntWritable one = new IntWritable(1);

  @Override
  protected void map(Object key, Text value, Context context) throws IOException, InterruptedException {
    StringTokenizer st = new StringTokenizer(value.toString());

    while (st.hasMoreTokens()) {
      word.set(st.nextToken().replaceAll("\\W", ""));
      context.write(word, one);
    }
  }
}
